cmd/limactl/copy.go

```diff
@@ -118,24 +161,67 @@ func copyAction(cmd *cobra.Command, args []string) error {
 		for _, inst := range instances {
 			sshOpts, err = sshutil.SSHOpts("ssh", inst.Dir, *inst.Config.User.Name, false, false, false, false)
 			if err != nil {
-				return err
+				return nil, nil, err
 			}
 		}
 	} else {
 		// Copying among multiple hosts; we can't pass in host-specific options.
 		sshOpts, err = sshutil.CommonOpts("ssh", false)
 		if err != nil {
-			return err
+			return nil, nil, err
 		}
 	}
-	sshArgs := sshutil.SSHArgsFromOpts(sshOpts)
+	sshArgs = sshutil.SSHArgsFromOpts(sshOpts)
 
-	sshCmd := exec.Command(arg0, append(sshArgs, scpArgs...)...)
-	sshCmd.Stdin = cmd.InOrStdin()
-	sshCmd.Stdout = cmd.OutOrStdout()
-	sshCmd.Stderr = cmd.ErrOrStderr()
-	logrus.Debugf("executing scp (may take a long time): %+v", sshCmd.Args)
+	return sshArgs, scpArgs, nil
+}
 
-	// TODO: use syscall.Exec directly (results in losing tty?)
-	return sshCmd.Run()
+func useRsync(args []string, verbose, recursive bool) ([]string, error) {
+	instances := make(map[string]*store.Instance)
+
+	var sshStr, instName string
+
+	rsyncFlags := []string{}
+	rsyncArgs := []string{}
+
+	if verbose {
+		rsyncFlags = append(rsyncFlags, "-v", "--progress")
+	} else {
+		rsyncFlags = append(rsyncFlags, "-q")
+	}
+
+	if recursive {
+		rsyncFlags = append(rsyncFlags, "-r")
+	}
+
+	for _, arg := range args {
+		path := strings.Split(arg, ":")
+		switch len(path) {
+		case 1:
+			inst, ok := instances[instName]
```

Comment:
How this will work? instances is an empty map created in this function.

I think we have only one case - `instance:path`, and it is common to both scp and rsync, so this check should be done first before we consider the copy tool. Input validation must always be done first.